### PR TITLE
RavenDB-13527 - Allow including timings to all queries when traffic watch is on, and make sure it's shown in the traffic watch

### PR DESCRIPTION
**Pull request by YonatanResh**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-13527/Allow-including-timings-to-all-queries-when-traffic-watch-is-on-and-make-sure-its-shown-in-the-traffic-watch

### Additional description

We added query timings property to `TrafficWatchHttpChange`.

### Type of change

- New feature

### How risky is the change?

- Low 

### Testing by Contributor

- It has been verified by manual testing

### UI work

- It requires further work in the Studio. Please mark the issue on YouTrack using `Studio Required` tag.

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Client/Documents/Changes/ChangeNotification.cs [hunk 1]
```diff
@@ -412,6 +413,7 @@ internal class TrafficWatchHttpChange : TrafficWatchChangeBase
         public long RequestSizeInBytes { get; set; }
         public long ResponseSizeInBytes { get; set; }
         public TrafficWatchChangeType Type { get; set; }
+        public QueryTimings Timings { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Changes/ChangeNotification.cs]: property should be called QueryTimings

File: src/Raven.Server/Documents/Queries/QueryMetadata.cs [hunk 2]
```diff
@@ -412,6 +413,15 @@ private void Build(BlittableJsonReaderObject parameters)
                 }
             }
 
+            if (TrafficWatchManager.HasRegisteredClients)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Queries/QueryMetadata.cs]: you cannot set it here, QueryMetadata is being cached. You need to handle that in 'SetupTimings' method IMO

File: src/Raven.Client/Documents/Queries/Timings/QueryTimings.cs [hunk 3]
```diff
@@ -1,10 +1,11 @@
 ﻿using System.Collections.Generic;
 using Raven.Client.Documents.Conventions;
 using Sparrow.Json;
+using Sparrow.Json.Parsing;
 
 namespace Raven.Client.Documents.Queries.Timings
 {
-    public class QueryTimings : IFillFromBlittableJson
+    public class QueryTimings : IFillFromBlittableJson, IDynamicJsonValueConvertible
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Queries/Timings/QueryTimings.cs]: use IDynamicJson and in TrafficWatchHttpChange just do `Timings?.ToJson()`

File: src/Raven.Client/Documents/Changes/ChangeNotification.cs [hunk 4]
```diff
@@ -425,6 +427,7 @@ public override DynamicJsonValue ToJson()
             json[nameof(RequestSizeInBytes)] = RequestSizeInBytes;
             json[nameof(ResponseSizeInBytes)] = ResponseSizeInBytes;
             json[nameof(Type)] = Type;
+            json[nameof(Timings)] = Timings;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Changes/ChangeNotification.cs]: do not write that property if Timings are null?

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 5]
```diff
@@ -147,6 +149,7 @@ private async Task Query(QueryOperationContext queryContext, OperationCancelToke
             try
             {
                 result = await Database.QueryRunner.ExecuteQuery(indexQuery, queryContext, existingResultEtag, token).ConfigureAwait(false);
+                AddQueryTimingsToTrafficWatch(indexQuery);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: can we add it after writing the response? probably after 'AddPagingPerformanceHint'

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs [hunk 6]
```diff
@@ -131,6 +132,7 @@ private async Task Query(QueryOperationContext queryContext, OperationCancelToke
             if (indexQuery.Metadata.HasFacet)
             {
                 await FacetedQuery(indexQuery, queryContext, token);
+                AddQueryTimingsToTrafficWatch(indexQuery);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: handle this inside 'FacetedQuery' ?

File: src/Raven.Client/Documents/Changes/ChangeNotification.cs (hunk 1 above)
Comment (YonatanResh) [reply src/Raven.Client/Documents/Changes/ChangeNotification.cs]: Done.

File: src/Raven.Server/Documents/Queries/QueryMetadata.cs (hunk 2 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Queries/QueryMetadata.cs]: Done.

File: src/Raven.Client/Documents/Queries/Timings/QueryTimings.cs (hunk 3 above)
Comment (YonatanResh) [reply src/Raven.Client/Documents/Queries/Timings/QueryTimings.cs]: Done.

File: src/Raven.Client/Documents/Changes/ChangeNotification.cs (hunk 4 above)
Comment (YonatanResh) [reply src/Raven.Client/Documents/Changes/ChangeNotification.cs]: Done.

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs (hunk 5 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: Done.

File: src/Raven.Server/Documents/Handlers/QueriesHandler.cs (hunk 6 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/Handlers/QueriesHandler.cs]: Done.

**Review by grisha-kotler — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]
Requires Studio work

**Review by ml054 — APPROVED** [review]

**Pull request merged**